File: p1/R/main.R
# Entry point for p1 R analysis.
# Paper: Prompt Quality x GenAI Response.
# Sub-commands:
#   Rscript main.R            -> design analysis (synthetic / future real data).
#   Rscript main.R pilot      -> Appendix-B pilot analysis on pilot_aioq_r_scores.csv.

source("src/00-utils.R")
source("src/01-data-prep.R")
source("src/02-analysis.R")
source("src/03-visualization.R")
source("src/04-pilot-analysis.R")
source("src/05-pilot-figures.R")

main_design <- function() {
  data <- load_data("../data/processed/aioq_r_scores.csv")
  cleaned <- prep(data)
  results <- run_rm_anova(cleaned)
  make_figures(results, "output/")
  invisible(results)
}

main_pilot <- function() {
  result <- run_pilot(
    input_csv = "../data/processed/pilot_aioq_r_scores.csv",
    output_dir = "output/",
    tex_out = "../latex/pilot-macros.tex"
  )
  plot_all(result, "output/")
  invisible(result)
}

if (sys.nframe() == 0) {
  args <- commandArgs(trailingOnly = TRUE)
  if (length(args) > 0 && args[1] == "pilot") main_pilot() else main_design()
}

File: p1/python/main.py
"""Entry point for p1 Python analysis.

Paper: Prompt Quality x GenAI Response.
Two sub-commands:
  python main.py pilot   — run the Appendix-B computational feasibility pilot.
  python main.py analyze — run the protocol-style summary on whatever sessions
                           JSONL is present in ../data/raw/.
"""

from __future__ import annotations

import sys
from pathlib import Path

from src.analysis import summarize
from src.data_prep import clean, load_raw
from src.pilot_run import run as run_pilot
from src.visualization import make_figures


HERE = Path(__file__).resolve().parent
RAW = HERE.parent / "data" / "raw"
PROCESSED = HERE.parent / "data" / "processed"


def _analyze() -> None:
    raw = load_raw(str(RAW / "sessions.jsonl"))
    df = clean(raw)
    summary = summarize(df)
    PROCESSED.mkdir(parents=True, exist_ok=True)
    df.to_csv(PROCESSED / "aioq_r_scores.csv", index=False)
    make_figures(summary, str(HERE / "output"))


def _pilot(n: int) -> None:
    summary = run_pilot(
        humaneval_path=str(RAW / "HumanEval.jsonl"),
        out_dir=str(PROCESSED),
        n_problems=n,
    )
    print("pilot done:", summary)


def main(argv: list[str]) -> int:
    cmd = argv[1] if len(argv) > 1 else "analyze"
    if cmd == "pilot":
        n = int(argv[2]) if len(argv) > 2 else 10
        _pilot(n)
    elif cmd == "analyze":
        _analyze()
    else:
        print(f"unknown command: {cmd}")
        return 2
    return 0


if __name__ == "__main__":
    sys.exit(main(sys.argv))
